wolvic/browser/vr/wvr_manager.cc

```diff
@@ -850,4 +875,62 @@ void WvrManager::UpdateLayerBounds(int16_t frame_index,
   CreateOrResizeWebXrSurface(source_size);
 }
 
+device::mojom::XREnvironmentBlendMode
+WvrManager::PickEnvironmentBlendModeForSession(
+    device::mojom::XRSessionMode session_mode) {
+  DCHECK(wvr_api_->get_system_state().enumerationCompleted);
+
+  const auto& display_state = wvr_api_->get_system_state().displayState;
+  auto supportsBlendMode =
+      [&display_state](const mozilla::gfx::VRDisplayBlendMode blend_mode) {
+        for (auto& supported_blend_mode : display_state.blendModes) {
+          if (supported_blend_mode ==
+              mozilla::gfx::VRDisplayBlendMode::_empty) {
+            return false;
+          }
+          if (supported_blend_mode == blend_mode) {
+            return true;
+          }
+        }
+        return false;
+      };
+
+  switch (session_mode) {
```

Comment:
Wouldn't be better to define a lamdba as getSuppotedBlendMode(device::mojom::XRSessionMode session_mode) and define the behavior based on that  ?

Reply:
you mean a lambda with the contents of this switch ?

Reply:
I've spent sometime trying to make this function clearer without much success, so probably not worth going further. We can apply some minor changes. 

If I've understood it correctly, we run NOTREACHED if the blend_modes contains "_empty"; couldn't we just make this a DCHECK at the beginning of the function and remove it from the supportBlendMode lambda ? 

If we remove the empty check, we could implement the supportBlendMode landa like this:

```
     auto supportsBlendMode =
      [&blend_modes](const mozilla::gfx::VRDisplayBlendMode mode) {
        return std::find(begin(blend_modes), end(blend_modes), mode) != end(blend_modes);
      };
```

I'll comment inline other changes that could make this function simpler.

Reply:
Hmm I don't get why you say the NOTREACHED is returned if the blend mode contains _empty, that is not correct. Also blend_modes is a C array not a vector